src/main/java/mate/academy/dao/impl/CarDaoImplJdbc.java

```diff
@@ -0,0 +1,206 @@
+import mate.academy.lib.Dao;
+import mate.academy.model.Car;
+import mate.academy.model.Driver;
+import mate.academy.model.Manufacturer;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImplJdbc implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getLong(1));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't create car "
+                    + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT cars.id, cars.model, cars.manufacturer_id, "
+                + "manufacturers.name, manufacturers.country FROM cars "
+                + "JOIN manufacturers ON cars.manufacturer_id = "
+                + "manufacturers.id WHERE cars.id = ? AND cars.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseDataFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get car by ID "
+                    + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.id, cars.model, cars.manufacturer_id, manufacturers.name, "
+                + "manufacturers.country FROM cars "
+                + "LEFT JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE taxi_service.cars.deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseDataFromResultSet(resultSet));
+            }
+            setCars(cars, connection);
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get all cars "
+                    + cars, e);
+        }
```

Comment:
```suggestion
        List<Car> cars = new ArrayList<>();
        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
                connection.prepareStatement(query)) {
            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                cars.add(parseDataFromResultSet(resultSet));
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Something went wrong. Can't get all cars "
                    + cars, e);
        }
        for (Car car : cars) {
            car.setDrivers(getCarDrivers(car.getId()));
        }
        return cars;
```
let's do the following - this way we will close the first connection and the first statement - and then open the new ones in getCarDrivers() method